internal/querywatcher/query_watcher.go

```diff
@@ -121,16 +121,26 @@ func (w *QueryWatcher) watchKeys(ctx context.Context, watchChan <-chan dstore.Wa
 }
 
 // processWatchEvent processes a single watch event.
-func (w *QueryWatcher) processWatchEvent(event dstore.WatchEvent) {
+func (w *QueryManager) processWatchEvent(event dstore.WatchEvent) {
+	// Iterate over the watchlist to go through the query string
+	// and the corresponding client connections to that query string
 	w.WatchList.Range(func(key, value interface{}) bool {
-		query := key.(DSQLQuery)
+		queryString := key.(string)
```

Comment:
For my understanding, why are we changing the key of the WatchList to a string? Isn't it possible to keep it as a `DSQLQuery` object? That would save us a ParseQuery call on every execution.

Reply:
This is precisely the change being made - https://github.com/DiceDB/dice/pull/456/
Hence I thought of combining this to prevent QWATCH tests from failing.

Reply:
Understood, thanks for the clarification!